### PR TITLE
Hide AggregateBody fields

### PR DESCRIPTION
**Pull request by CjS77**
AggregateBody's filed must be private becuase there are side effects
to adding inputs, outputs and kernels (the struct is marked as unsorted).

This PR makes that change, introduced read-only getters, and fixes
all the places that were accessing the fields directly.

### CODE REVIEW

File: base_layer/core/src/blocks/aggregated_body.rs [hunk 1]
```diff
@@ -69,6 +69,27 @@ impl AggregateBody {
...
+        &self.outputs
+    }
+
+    /// Provide read-only access to the kernel list
+    pub fn kernels(&self) -> &Vec<TransactionKernel> {
+        &self.kernels
+    }
+
+    /// Available for tests only. Get a mutable reference to the inputs
+    #[cfg(test)]
```
Comment (CjS77) [inline base_layer/core/src/blocks/aggregated_body.rs]: I'd love to **NOT** have this function (at least it's test-only), so open to suggestions for removing its requirement at (1)

File: base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs [hunk 2]
```diff
@@ -292,8 +292,8 @@ mod test {
         let mut tx5 = tx!(MicroTari(5_000), fee:MicroTari(50), inputs:3, outputs:1).0;
         let mut tx6 = tx!(MicroTari(5_000), fee:MicroTari(75), inputs: 2, outputs: 1).0;
         // tx1 and tx5 have a shared input. Also, tx3 and tx6 have a shared input
-        tx5.body.inputs[0] = tx1.body.inputs[0].clone();
-        tx6.body.inputs[1] = tx3.body.inputs[1].clone();
+        tx5.body.inputs_mut()[0] = tx1.body.inputs()[0].clone();
+        tx6.body.inputs_mut()[1] = tx3.body.inputs()[1].clone();
```
Comment (CjS77) [inline base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs]: (1) This is the only place `inputs_mut` is used; it'd be great not to have to have this fn

File: base_layer/core/src/blocks/aggregated_body.rs (hunk 1 above)
Comment (philipr-za) [reply base_layer/core/src/blocks/aggregated_body.rs]: Afraid I cannot think of a good way to not have this to facilitate the tests at (1)

File: base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs (hunk 2 above)
Comment (neonknight64) [reply base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs]: We could use `#[cfg(test)]` to only make the function available for testing.

Comment (CjS77) [reply base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs]: That's what we're doing right now.

**Pull request merged**